cmd/krew/cmd/index.go

```diff
@@ -62,14 +67,67 @@ var indexAddCmd = &cobra.Command{
 	Example: "kubectl krew index add default " + constants.IndexURI,
 	Args:    cobra.ExactArgs(2),
 	RunE: func(_ *cobra.Command, args []string) error {
-		return indexoperations.AddIndex(paths.IndexBase(), args[0], args[1])
+		return indexoperations.AddIndex(paths, args[0], args[1])
 	},
 }
 
+var indexDeleteCmd = &cobra.Command{
+	Use:   "remove",
+	Short: "Remove a configured index",
+	Long: `Remove a configured index repository which is used to discover and
+install plugins from.
+
+If there are plugins installed from the specified index, the index cannot be
+removed (unless --force) is used. Removing indexes while there are plugins
+installed from them is not supported behavior.`,
+	Args: cobra.ExactArgs(1),
+	RunE: indexDelete,
+}
+
+func indexDelete(_ *cobra.Command, args []string) error {
+	name := args[0]
+
+	pl, err := installation.InstalledPluginsFromIndex(environment.MustGetKrewPaths().InstallReceiptsPath(), name)
+	if err != nil {
+		return errors.Wrap(err, "failed querying plugins installed from the index")
+	}
+	klog.V(4).Infof("Found %d plugins from index", len(pl))
+
+	if len(pl) > 0 && !*forceIndexDelete {
+		var names []string
+		for _, v := range pl {
+			names = append(names, v.Name)
+		}
+
+		warning := color.New(color.FgRed, color.Bold).Sprint("WARNING:")
```

Comment:
Would this be worthwhile putting somewhere? There's a couple places where something like this exists `boldRed := color.New(color.FgRed, color.Bold).SprintFunc()` and is called `boldRed("WARNING")`. I'm not even sure where it would go and this is pretty minor, just thought I'd bring it up.

Reply:
+1

Reply:
Done. `internal.PrintWarning` method created and existing uses migrated.